src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -495,6 +496,11 @@ void SetValueWithTrace(DependencyProperty property, object? value, DependencyPro
 
 		private void InnerSetValue(DependencyProperty property, object? value, DependencyPropertyValuePrecedences precedence, DependencyPropertyDetails? propertyDetails, bool isPersistentResourceBinding)
 		{
+			if (!FeatureConfiguration.DependencyProperty.DisableThreadingCheck)
+			{
+				NativeDispatcher.CheckThreadAccess();
```

Comment:
TODO:

This is failing these tests:

![image](https://github.com/user-attachments/assets/d94ef66f-600d-4489-8d9c-4de3e5e333e3)